kindelia_core/src/node.rs

```diff
@@ -1536,21 +1547,14 @@ impl<C: ProtoComm, S: BlockStorage> Node<C, S> {
     self.send_blocks_to(addrs, true, blocks, 3);
   }
 
-  pub fn load_blocks(&mut self) {
+  pub fn load_blocks(&mut self) -> Result<(), NodeError> {
     self.storage.disable();
     let storage = self.storage.clone();
-    storage.read_blocks(|(block, file_path)| match block {
-      Some(block) => {
-        self.add_block(&block.hashed());
-      }
-      None => {
-        eprintln!(
-          "WARN: Could not load block from file '{}'",
-          file_path.display()
-        );
-      }
-    });
+    storage
+      .read_blocks(|(block, _file_path)| self.add_block(&block.hashed()))
+      .map_err(|e| NodeError::BlockStorage(e))?;
```

Comment:
As `transparent` is being used there in `NodeError::BlockStorage`, I think it is not necessary to do this `map_err`

Reply:
good point, I changed it.